docs/using.md

```diff
@@ -0,0 +1,282 @@
+
+Let's say there were three streams and two replicas, you'd get the running total of two streams for one replica, and the running total for the third stream in the other replica.
+
+In most practical scenarios you'd want to track a running total per stream (say, total g-forces per race car), or perhaps for some variables a running total across all streams. Each of these scenarios is described in the followng sections.
+
+## Tracking running totals per stream
+
+Sometimes you might want to calculate a running total of a variable for a stream. For example, the total g-force a racing car is exposed to. If you use a global variable you'll lose the stream context. All streams will add to the value potentially, and each replica will also have its own instance of the global, further confusing matters.
+
+The solution is to use the stream context to preserve a running total for that stream only. To do this you can use the `stream_id` of a stream to identify its data. Consider the following example:
+
+```python
+...
+g_running_total_per_stream = {}
+
+def callback_handler (stream_consumer: qx.StreamConsumer, data: qx.TimeseriesData):
+
+    if stream_consumer.stream_id not in g_running_total_per_stream:
+        g_running_total_per_stream[stream_consumer.stream_id] = 0
+    
+    ...
+
+    g_running_total_per_stream[stream_consumer.stream_id] += some_value
+
+...
+```
+
+The key point here is that data is tracked per stream context. You keep running totals on a per-stream basis by using the stream ID, `stream_consumer.stream_id` to index a dictionary containing running totals for each stream.
+
+!!! note
+
+    A stream will only ever be processed by one replica.
+
+## Tracking running totals across multiple streams
+
+Sometimes you want to track a running total across all streams in a topic. The problem is that when you scale using replicas, there is no way to share data between all replicas in a consumer group. 
+
+The solution is to write the running total per stream (with stream ID) to an output topic. You can then have another processor in the pipeline to calculate total values from inbound messages.
+
+```python
+...
+g_running_total_per_stream = {}
+
+def callback_handler (stream_consumer: qx.StreamConsumer, data: qx.TimeseriesData):
+
+    if stream_consumer.stream_id not in g_running_total_per_stream:
+        g_running_total_per_stream[stream_consumer.stream_id] = 0
+    
+    ...
+
+    g_running_total_per_stream[stream_consumer.stream_id] += some_value
+    data.add_value("RunningTotal", g_running_total_per_stream[stream_consumer.stream_id])
+
+    topic_producer.get_or_create_stream(stream_consumer.stream_id).timeseries.publish(data)
+...
+```
+
+In this case the running total is published to its own stream in the output topic. The next service in the data processing pipeline would be able to sum all running totals across all streams in the output topic.
+
+## Handling system restarts and crashes
```

Comment:
I would add some explanation around the fact that any in-memory data will be lost on service crash or shutdown. This can be mitigated by using LocalFileStorage etc.

Reply:
Hopefully I've now clarified this section a bit.